src/lib.rs

```diff
@@ -32,6 +33,91 @@ struct Metadata {
     inner: ChdMetadata,
 }
 
+#[pyclass]
+struct Header {
+    inner: ChdHeader,
+}
+
+#[pymethods]
+impl Header {
+    pub fn is_compressed(&self) -> bool
+    {
+        self.inner.is_compressed()
+    }
+
+    pub fn meta_offset(&self) -> Option<u64>
+    {
+        self.inner.meta_offset()
+    }
+
+    pub fn flags(&self) -> Option<u32>
+    {
+        self.inner.flags()
+    }
+
+    pub fn hunk_count(&self) -> u32
+    {
+        self.inner.hunk_count()
+    }
+
+    pub fn hunk_size(&self) -> u32
+    {
+        self.inner.hunk_size()
+    }
+
+    pub fn logical_bytes(&self) -> u64
+    {
+        self.inner.logical_bytes()
+    }
+
+    pub fn unit_bytes(&self) -> u32
+    {
+        self.inner.unit_bytes()
+    }
+
+    pub fn unit_count(&self) -> u64
+    {
+        self.inner.unit_count()
+    }
+
+    pub fn has_parent(&self) -> bool
+    {
+        self.inner.has_parent()
+    }
+
+    pub fn len(&self) -> u32 {
+        self.inner.len()
+    }
+
+    pub fn get_sha1(&self) -> &[u8] {
```

Comment:
Can you rename this to `sha1` to be consistent with the other methods, and have it return `Option<&[u8]>`? That way you can return `Some(h)` for V5, and `None` for non-v5.

Reply:
V3 and V4 also have sha1